app/main.py

```diff
@@ -1,23 +1,40 @@
-import init_django_orm  # noqa: F401
+from django.db.models import QuerySet, Q, F
 
-from django.db.models import QuerySet
+import init_django_orm  # noqa: F401
+from db.models import LoyaltyProgram, \
+    LoyaltyProgramParticipant, \
+    Customer
 
 
 def all_loyalty_program_names() -> QuerySet:
-    pass
+    return LoyaltyProgram. \
+        objects.values_list("name", "bonus_percentage")
 
 
 def not_active_customers() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects. \
+        filter(last_activity__gte="2021-01-01",
+               last_activity__lt="2022-01-01"). \
```

Comment:
you can use __year=2021

Reply:
This function returns all names of loyalty participants who were active in the period from 2021-01-01 to 2022-01-01.
Period, like BETWEEN
I need to correct it like last_activity__year=2021?
Or let it remain as it is?

Reply:
Just try it)